### PR TITLE
add probablity dist functions

### PR DESCRIPTION
**Pull request by Marcos-cat**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -36,6 +36,7 @@ Functions list (see in-code documentation for more detail)
 | `Qqp` | Quaternion product |
 | `RQuat` | Create 3D rotation quaternions |
 | `QRot` | Rotate a 3D vector array using a quaternion array |
+| `Fact` | Factorial of n |
```
Comment (Omnikar) [inline README.md]:
```suggestion
| `Fact` | Pervasive factorial |
```

File: lib.ua [hunk 2]
```diff
@@ -244,6 +247,28 @@ BoxMuller ← ∩×⤙°∠×τ:√×¯2ₙe¬
 # ? Size StandardDeviation
 Gaussian ← ÷/+⊸♭ⁿ:e¯÷2°√÷:⌵⊞ℂ.-÷2-1⟜⇡
 
+# ? p n X
+BinomPmf ← ××⊃(∵⧅>|ⁿ⊙⋅∘|ⁿ⊓-¬) ⤚⋅:
+# ? p n X
+BinomCmf ← /+BinomPmf⊓∩¤(⇡+1)
```
Comment (Omnikar) [inline lib.ua]:
```suggestion
# Binomial distribution
# Probability that n Bernoulli trials each with success probability p will amount to X successes
# ? p n X
BinomPmf ← ××⊃(∵⧅>|ⁿ⊙⋅∘|ⁿ⊓-¬) ⤚⋅:
# Cumulative binomial distribution
# Probability that n Bernoulli trials each with success probability p will amount to X or fewer successes
# ? p n X
BinomCmf ← /+BinomPmf⊓∩¤(⇡+1)
```

File: README.md [hunk 3]
```diff
@@ -47,6 +48,14 @@ Functions list (see in-code documentation for more detail)
 | `DistRand` | Seeded random indices from a distribution array |
 | `BoxMuller` | Box-muller transform: generate pairs of normally distributed values given pairs of uniformly distributed values |
 | `Gaussian` | Generate a 2 dimensional square Gaussian kernel |
+| `BinomPmf` | Binomial distribution |
+| `BinomCmf` | Cumulative binomial distribution |
+| `PoissonPmf` | Poisson distribution |
+| `PoissonCmf` | Cumulative poisson distribution |
+| `GeomPmf` | Geometric distribution (0-indexed) |
+| `GeomCmf` | Cumulative geometric distribution (0-indexed) |
```
Comment (Omnikar) [inline README.md]:
```suggestion
| `GeomPmf` | Geometric distribution (0-indexed) |
| `GeomCmf` | Cumulative geometric distribution (0-indexed) |
| `PoissonPmf` | Poisson distribution |
| `PoissonCmf` | Cumulative Poisson distribution |
```

File: lib.ua [hunk 4]
```diff
@@ -244,6 +247,28 @@ BoxMuller ← ∩×⤙°∠×τ:√×¯2ₙe¬
...
+
+# Number of failures before the first success
+# ? p X
+GeomPmf ← ×˜ⁿ¬’
+# Number of failures before the first success
+# ? p X
+GeomCmf ← ⍜¬˜ⁿ⊙(+1)
```
Comment (Omnikar) [inline lib.ua]:
```suggestion
# Geometric distribution
# Probability of X failures before the first success of repeated trials with success probability p
# ? p X
GeomPmf ← ×˜ⁿ¬’
# Cumulative geometric distribution
# Probability of X or fewer failures before the first success of repeated trials with success probability p
# ? p X
GeomCmf ← ⍜¬˜ⁿ⊙(+1)
```

File: lib.ua [hunk 5]
```diff
@@ -244,6 +247,28 @@ BoxMuller ← ∩×⤙°∠×τ:√×¯2ₙe¬
...
+
+# ? λ X
+PoissonPmf ← ÷⊙×⊃(⋅Fact|˜ⁿ|ⁿ⊙e¯)
+# ? λ X
+PoissonCmf ← /+PoissonPmf⊓¤(⇡+1)
```
Comment (Omnikar) [inline lib.ua]:
```suggestion
# Poisson distribution
# Probability of X occurrences of an unlikely event over many trials with expected value λ
# ? λ X
PoissonPmf ← ÷⊙×⊃(⋅Fact|˜ⁿ|ⁿ⊙e¯)
# Cumulative Poisson distribution
# Probability of X or fewer occurrences of an unlikely event over many trials with expected value λ
# ? λ X
PoissonCmf ← /+PoissonPmf⊓¤(⇡+1)
```

File: README.md (hunk 3 above)
Comment (Omnikar) [reply README.md]: I'd forgotten to mention that since `Poisson` is a proper noun it needs to be capitalized.

File: lib.ua [hunk 6]
```diff
@@ -244,6 +247,28 @@ BoxMuller ← ∩×⤙°∠×τ:√×¯2ₙe¬
...
+
+# ? σ μ X
+NormalPdf ← ÷⊃(×√×2π|ⁿ⊙e÷2¯°√÷⊙-)
+# ? σ μ X
+NormalCdf ← ÷2+1Erf÷×√2⊙-
```
Comment (Omnikar) [inline lib.ua]:
```suggestion
# Normal distribution
# Probability density of a gaussian/bell curve with mean μ and standard deviation σ
# ? σ μ X
NormalPdf ← ÷⊃(×√×2π|ⁿ⊙e÷2¯°√÷⊙-)
# Cumulative normal distribution
# Probability that a normally distributed random variable with mean μ and standard deviation σ is less than X
# ? σ μ X
NormalCdf ← ÷2+1Erf÷×√2⊙-
```

**Comment by Marcos-cat** [comment]
what if `BinomCmf` mentions that it is not pervasive for the 3rd argument, and if this behavior is desired, use `each`? Using `each` by default seems like a bad call in cases where only the probability or number of trials is changing.

**Comment by Omnikar** [comment]
That's fine.

**Pull request merged**